src/distilabel/steps/filtering/minhash.py

```diff
@@ -0,0 +1,313 @@
+
+    Args:
+        texts: List of documents to be tokenized.
+        n: The size of the ngrams, defaults to 1 (single letters).
+
+    Returns:
+        Iterable[Set[bytes]]: List with the set of tokens for each document.
+    """
+
+    return [
+        {"".join(ngram).encode("utf-8") for ngram in ngrams(text, n=n)}
+        for text in texts
+    ]
+
+
+# NOTE: This class must be used together with the `MinHashLSH` class.
+# We return the `hashvalues` to reproduce the MinHash objects, but we also need
+# the seed, so the seed used for the MinHash objects must be kept to be grabbed
+# for the next class. We could also pass it as part of the dict, but there's no point.
+# Also, instead of returning the values, we could be saving them as artifacts,
+# This still needs to be studied.
+class MinHash(Step):
+    """Creates the components for a `MinHash` object to deduplicate texts.
+
+    From `datasketch` documentation:
+    Estimates the Jaccard similarity (resemblance) between sets of arbitrary sizes in linear
+    time using a small and fixed memory space.
+
+    Note:
+        We only keep the hashvalues, as using those values together with the seed
+        we can reproduce the `MinHash` objects. The `MinHashLSH` will recreate those internally.
+
+    Attributes:
+        column: the column to deduplicate. Defaults to `text`.
+        num_perm: the number of permutations to use. Defaults to `128`.
+        seed: the seed to use for the MinHash. Defaults to `1`.
+        tokenizer: the tokenizer to use. Available ones are `words` or `ngrams`.
+            If `words` is selected, it tokenize the text into words using nltk's
+            word tokenizer. `ngram` estimates the ngrams (together with the size
+            `n`) using. Defaults to `ngrams`.
+        n: the size of the ngrams to use. Only relevant if `tokenizer="ngrams"`. Defaults to `1`.
+
+    Input columns:
+        - text (`str`): the texts to obtain the hashes for.
+
+    Output columns:
+        - hashvalues (`List[int]`): hash values obtained for the algorithm.
+
+    Categories:
+        - filtering
+
+    References:
+        - [`datasketch documentation`](https://ekzhu.com/datasketch/minhash.html#minhash)
+        - [Identifying and Filtering Near-Duplicate Documents](https://cs.brown.edu/courses/cs253/papers/nearduplicate.pdf)
+
+    Examples:
+
+        Create MinHash objects for a list of texts to be deduplicated:
+
+        ```python
+        texts: List[str] = [
+            "This is a test document.",
+            "This document is a test.",
+            "Test document for duplication.",
+            "Document for duplication test.",
+            "This is another unique document."
+        ]
+        from distilabel.steps import MinHash
+        minhasher = MinHash(tokenizer="ngrams", n=3)
+        minhasher.load()
+        result = next(hasher.process([{"text": t} for t in texts]))
+        ```
+    """
+
+    column: str = "text"
```

Comment:
why do we need this attribute? We should handle these cases using `input_mappings`

Reply:
right, there's no need, I was using `input_mappings` anyway...